File: 20495_corejava/Day2Assignments/Truck.java
package com.sonata;

public class Truck extends Car{
	
	int weight;
	double discount;
	Truck(){}
	Truck(int s,double r,String c){
		super(s,r,c);
	}
	public void getSalePrice() {
		if(weight>2000)
		{
			discount=(0.1*regularPrice);
		}
		else
		{
		    discount=(0.2*regularPrice);
		}
		System.out.println(regularPrice-discount);
	}
	
//	public static void main(String[] args) {
//		
//		Truck t1=new Truck();
//		t1.weight=2000;
//		t1.speed=60;
//		t1.color="Red";
//	    t1.regularPrice=30000;
//	    t1.getSalePrice();
//	    t1.display();
//	}

}

File: TypeScriptAssignment/Truck.ts
import {Car} from './Car';
class Truck extends Car{

    weight:number;

    constructor(speed:number , regularPrice:number , color:string){
        super(speed,regularPrice,color);
    }

    getSalePrice(weight:number){
        if(Math.max(weight,2000)){
            this.salePrice=this.regularPrice-(10/100*this.regularPrice);
        }
        else{
            this.salePrice=this.regularPrice-(20/100*this.regularPrice);
        }

    }


}

var t=new Truck(100,200000,"Black");
t.getSalePrice(2000);
t.display();